src/corehost/common/pal.unix.cpp

```diff
@@ -169,6 +169,30 @@ bool pal::get_default_servicing_directory(string_t* recv)
 }
 
 #if defined(__APPLE__)
+bool pal::get_os_moniker(os_moniker_t* moniker)
+{
+    char str[256];
+
+    size_t size = sizeof(str);
+    int ret = sysctlbyname("kern.osrelease", str, &size, NULL, 0);
+    if (ret != 0)
+    {
+        return false;
+    }
+
+    string s(str, size);
+    size_t pos = s.find('.');
+    if (pos == string::npos)
+    {
+        return false;
+    }
+
+    *moniker = (os_moniker_t) stoi(s.substr(0, pos));
+    return true;
+}
+#endif
```

Comment:
You can remove the "#endif" and the following "#ifdef" since the _APPLE_ block is continuous.